integration/common/openlineage/common/provider/dbt.py

```diff
@@ -0,0 +1,202 @@
+import json
+import yaml
+import os
+import uuid
+from typing import List, Tuple, Dict, Optional
+
+import attr
+
+from openlineage.client.facet import DataSourceDatasetFacet, SchemaDatasetFacet, SchemaField, \
+    SqlJobFacet
+from openlineage.common import __version__
+from openlineage.client.run import RunEvent, RunState, Run, Job, Dataset
+from openlineage.common.utils import get_from_nullable_chain
+
+PRODUCER = f"openlineage-dbt/{__version__}"
+
+
+@attr.s
+class DbtRun:
+    started_at: str = attr.ib()
+    completed_at: str = attr.ib()
+    status: str = attr.ib()
+    inputs: List[Dict] = attr.ib()
+    output: Dict = attr.ib()
+    job_name: str = attr.ib()
+    namespace: str = attr.ib()
+    run_id: str = attr.ib(factory=lambda: str(uuid.uuid4()))
+
+
+@attr.s
+class DbtEvents:
+    starts: List[RunEvent] = attr.ib()
+    completes: List[RunEvent] = attr.ib()
+
+
+class DbtArtifactProcessor:
+    def __init__(self, project: str = 'dbt_project.yaml'):
+        self.dir = os.path.abspath(os.path.dirname(project))
+        self.project = self.load_yaml(project)
+        self.namespace = ""
+
+    def parse(self, target: Optional[str] = None) -> DbtEvents:
+        manifest = self.load_manifest(
+            os.path.join(self.dir, self.project['target-path'], 'manifest.json')
+        )
+        run_result = self.load_run_results(
+            os.path.join(self.dir, self.project['target-path'], 'run_results.json')
+        )
+
+        profile_dir = run_result['args']['profiles_dir']
+
+        profile = self.load_yaml(
+            os.path.join(profile_dir, 'profiles.yaml')
+        )[self.project['profile']]
+
+        if target:
+            profile = profile['outputs'][target]
+        else:
+            profile = profile['outputs'][profile['target']]
+
+        self.set_namespace(profile)
+
+        runs = self.parse_artifacts(manifest, run_result)
+
+        start_events, end_events = [], []
+        for run in runs:
+            start, end = self.to_openlineage_events(run)
```

Comment:
We should handle the `ValueError` thrown by `to_openlineage_events()`. We either want to log the error and continue, or exit out of the loop.